packages/datadog-plugin-aws-sdk/src/base.js

```diff
@@ -0,0 +1,152 @@
+'use strict'
+
+const Tags = require('opentracing').Tags
+const analyticsSampler = require('../../dd-trace/src/analytics_sampler')
+const Plugin = require('../../dd-trace/src/plugins/plugin')
+const { storage } = require('../../datadog-core')
+
+class BaseAwsSdkPlugin extends Plugin {
+  static get serviceIdentifier () {
+    const id = this.name.toLowerCase()
+    Object.defineProperty(this, 'serviceIdentifier', {
```

Comment:
This caches the value on the extended class after the first access. `defineProperty` is needed because we're defining it as a getter starting on line 9. A similar thing is done with the instance version of this below.

Reply:
That being said, I think only the instance one is used over and over again, so I can just cache that one.

Reply:
If it just uses the name, why does it need any caching?